dot/state/storage.go

```diff
@@ -162,7 +162,49 @@ func (s *StorageState) StoreInDB(root common.Hash) error {
 		return errTrieDoesNotExist(root)
 	}
 
-	return StoreTrie(s.baseDB, s.tries[root])
+	err := StoreTrie(s.baseDB, s.tries[root])
+	if err != nil {
+		return err
+	}
+
+	// notify subscribers of database changes
+	for _, sub := range s.subscriptions {
+		subRes := &SubscriptionResult{
+			Hash: root,
+		}
+		if len(sub.Filter) == 0 {
+			// no filter, so send all changes
+			ent := s.tries[root].Entries()
+			for k, v := range ent {
+				if k != ":code" {
+					// todo, currently we're ignoring :code since this is a lot of data
+					kv := &KeyValue{
+						Key:   common.MustHexToBytes(fmt.Sprintf("0x%x", k)),
+						Value: v,
+					}
+					subRes.Changes = append(subRes.Changes, *kv)
+				}
+			}
+
```

Comment:
this seems to send all the entries in the trie, not just the ones that are changed? am I mistaken?

Reply:
Right, I wasn't sure about this, however when I was testing with Substrate node, using key 0x26aa394eea5630e07c48ae0c9558cef70a98fdbe9ce6c55837576c60c7af3850 (System EventCount) I saw that it was responding with a value every block, even though the value remained the same (1).  So I figured that I should replicate the same behavior.